NumberEntryWidget increment/decrement values are now configurable

WithersChat opened the pull request:
- NumberEntryWidget now reads from config at creation.
- 3rd page of Terminal settings is used for in-game config changes.
- Added AETextDisplayWidget, a wrapping text display widget for the sole purpose of having text with wraparound that extends AbstractWidget and can thus be seamlessly used in a paged UI.
- Added IntegerTextField, an extension of AETextField that can only contain integers or an empty string. Also has methods to get and set its value as and from an int, for cleaner use in TerminalSettingsScreen code.
- Added methods to create AETextDisplayWidget and IntegerTextField to TerminalSettingsScreen and WidgetContainer
- The config entries are stored as an array for ease of access in other parts of code.
- Relevant lang entries added to GuiText, datagen has been executed

WithersChat committed:
NumberEntryWidget increment/decrement values are now configurable

- NumberEntryWidget now reads from config at creation.
- 3rd page of Terminal settings is used for in-game config changes.
- Added AETextDisplayWidget, a wrapping text display widget for the sole purpose of having text with wraparound that extends AbstractWidget and can thus be seamlessly used in a paged UI.
- Added IntegerTextField, an extension of AETextField that can only contain integers or an empty string. Also has methods to get and set its value as and from an int, for cleaner use in TerminalSettingsScreen code.
- Added methods to create AETextDisplayWidget and IntegerTextField to TerminalSettingsScreen and WidgetContainer
- The config entries are stored as an array for ease of access in other parts of code.
- Relevant lang entries added to GuiText, datagen has been executed

WithersChat commented:
<img width="1262" height="359" alt="image" src="https://github.com/user-attachments/assets/aebab2e5-3e93-4655-ad05-d28204367f31" />
Spotless decided not to cooperate so I hope I didn't make errors on that front...

JuiceyBeans commented on src/main/java/appeng/client/gui/me/common/TerminalSettingsScreen.java:
```diff
@@ -1,7 +1,9 @@
 package appeng.client.gui.me.common;
 
+import appeng.client.gui.widgets.*;
```
Change this to the actual class imports so that Spotless doesn't complain about wildcard imports

WithersChat committed:
A couple fixes

- AETextDisplayWidget doesn't make a sound when clicked anymore.
- Rearranged the new Terminal Settings page to have better button size and spacing.

WithersChat committed:
Cleaned up imports

- No more wildcards

WithersChat committed:
Merge branch 'forge/1.20.1' into configurable-number-widget

Ghostipedia committed:
Merge branch 'forge/1.20.1' into configurable-number-widget

WithersChat committed:
Spotless

WithersChat committed:
Merge branch 'configurable-number-widget' of https://github.com/WithersChat/Applied-Energistics-2-cosmolite into configurable-number-widget

Ghostipedia committed:
Merge branch 'forge/1.20.1' into configurable-number-widget
